services/task-svc/internal/events/events.go

```diff
@@ -0,0 +1,240 @@
+	event := hwes.NewBaseEvent(BedUpdated)
+	if err := event.SetJsonData(eventData); err != nil {
+		return hwes.Event{}, err
+	}
+	return event, nil
+}
+
+func NewBedMovedToAnotherRoomEvent(roomId string) (hwes.Event, error) {
+	eventData := BedMovedToAnotherRoomEvent{
+		RoomId: roomId,
+	}
+	event := hwes.NewBaseEvent(BedUpdated)
+	if err := event.SetJsonData(eventData); err != nil {
+		return hwes.Event{}, err
+	}
+	return event, nil
+}
+
+func NewTaskAssignedEvent(userId string) (hwes.Event, error) {
+	eventData := TaskAssignedEvent{
+		UserId: userId,
+	}
+	event := hwes.NewBaseEvent(TaskAssigned)
+	if err := event.SetJsonData(eventData); err != nil {
+		return hwes.Event{}, err
+	}
+	return event, nil
+}
+
+func NewTaskSelfAssignedEvent(userId string) (hwes.Event, error) {
+	eventData := TaskSelfAssignedEvent{
+		UserId: userId,
+	}
+	event := hwes.NewBaseEvent(TaskSelfAssigned)
+	if err := event.SetJsonData(eventData); err != nil {
+		return hwes.Event{}, err
+	}
+	return event, nil
+}
+
+func DispatchBedCreatedEvent(ctx context.Context, bedId uuid.UUID, bedName string) error {
+	if !feature.IsEventSourcingEnabled() {
+		return nil
+	}
+
+	log := zlog.Ctx(ctx)
+	event, err := NewBedCreatedEvent(bedId, bedName)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("event", BedCreated).
+			Msg("could not create event")
+		return err
+	}
+
+	err = hwes.AppendEvents(ctx, bedId.String(), event)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("event", event.Type).
+			Msg("could not dispatch event")
+	}
+	return err
+}
+
+func DispatchBedUpdatedEvent(ctx context.Context, bedId uuid.UUID, bedName string) error {
+	if !feature.IsEventSourcingEnabled() {
+		return nil
+	}
+
+	log := zlog.Ctx(ctx)
+	event, err := NewBedUpdatedEvent(bedName)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("event", BedUpdated).
+			Msg("could not create event")
+		return err
+	}
+
+	err = hwes.AppendEvents(ctx, bedId.String(), event)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("event", event.Type).
+			Msg("could not dispatch event")
+	}
+	return err
+}
+
+func DispatchBedMovedToAnotherRoomEvent(ctx context.Context, bedId uuid.UUID, roomId string) error {
+	if !feature.IsEventSourcingEnabled() {
+		return nil
+	}
+
+	log := zlog.Ctx(ctx)
+	event, err := NewBedMovedToAnotherRoomEvent(roomId)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("event", BedMovedToAnotherRoom).
+			Msg("could not create event")
+		return err
+	}
+
+	err = hwes.AppendEvents(ctx, bedId.String(), event)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("event", event.Type).
+			Msg("could not dispatch event")
+	}
+	return err
+}
+
+func DispatchBedDeletedEvent(ctx context.Context, bedId uuid.UUID) error {
+	if !feature.IsEventSourcingEnabled() {
+		return nil
+	}
+
+	log := zlog.Ctx(ctx)
+	err := hwes.AppendEvents(ctx, bedId.String())
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("event", BedDeleted).
+			Msg("could not dispatch event")
+	}
```

Comment:
You are calling `AppendEvents` without passing an event, is that intended?
The Event is also not showing up in my eventstore db stream

Reply:
Not intended! Thanks.